src/player.cpp

```diff
@@ -512,17 +514,17 @@ void player::action_taken()
 void player::update_morale()
 {
     // Decay existing morale entries.
-    for( size_t i = 0; i < morale.size(); i++ ) {
+    for( size_t i = morale.size() - 1; i > 0; i-- ) {
```

Comment:
`size_t` can be unsigned. From what I've seen, it is signed in Windows MinGW (for example, CodeBlocks), but unsigned on GNU/Linux when compiled with GCC.

When unsigned, `morale.size() - 1` will wrap around when `morale.size() == 0`, then `i > 0` will be true and `morale[i].proceed()` will be called. This causes a crash.